client/my-sites/checkout/checkout/checkout.jsx

```diff
@@ -341,19 +343,46 @@ export class Checkout extends React.Component {
 		if ( this.props.isNewlyCreatedSite && receipt && isEmpty( receipt.failed_purchases ) ) {
 			const siteDesignType = get( selectedSite, 'options.design_type' );
 			const hasGoogleAppsInCart = cartItems.hasGoogleApps( cart );
+			const hasConciergeSessionInCart = cartItems.hasConciergeSession( cart );
 
+			// Handle the redirect path after a purchase of GSuite or Concierge Session
 			// The onboarding checklist currently supports the blog type only.
-			if ( hasGoogleAppsInCart && domainReceiptId && 'store' !== siteDesignType ) {
+			if (
+				( hasConciergeSessionInCart || ( hasGoogleAppsInCart && domainReceiptId ) ) &&
+				'store' !== siteDesignType
+			) {
 				analytics.tracks.recordEvent( 'calypso_checklist_assign', {
 					site: selectedSiteSlug,
 					plan: 'paid',
 				} );
+
+				if ( config.isEnabled( 'upsell/concierge-session' ) && hasConciergeSessionInCart ) {
+					return `/checklist/${ selectedSiteSlug }`;
```

Comment:
Did you mean this for the condition above?
```
if (
    ( hasConciergeSessionInCart || ( hasGoogleAppsInCart && domainReceiptId ) ) &&
    'store' !== siteDesignType
) {
```
It would be nice to use a single function to check whether the checklist should be shown. But currently that function uses different logic and so we can't use it here as is. I'm also not entirely familiar with all the criteria that we want to check to determine if we should show the checklist. I think changing that code is a bit out of scope for this particular PR/project.